### PR TITLE
Update to pytorch==1.0.0

### PR DESCRIPTION
**Pull request by avinashsai**
Hi,
I reverted back those changes as discussed with .eval() and torch.no_grad() additions.
I also created a variable numbatches so that we don't have to compute it again and again after every 50 batches for each epoch.

### CODE REVIEW

**Comment by rasbt** [comment]
Thanks! A maybe even betters solution would be to replace

`len(train_dataset)//batch_size` (or `num_batches`) by `len(train_loader)` as this would then also account for scenarios where the training set is not evenly divisible into the specified number of batches

**Comment by avinashsai** [comment]
I don't think it would make much difference even if it is not divisible. we will get same value in either cases for numbatches.

**Comment by avinashsai** [comment]
Sure i will change it len(train_loader)

**Comment by avinashsai** [comment]
Changed numbatches = len(train_loader)

**Comment by avinashsai** [comment]
Any further updates to be made??

**Comment by rasbt** [comment]
I think it would be better to change ` F.log_softmax(logits, dim=1)` back to `F.softmax(logits, dim=1)` because the log(softmax) is not used in the cross entropy loss and probas may be more useful during inference then log probas

**Comment by avinashsai** [comment]
I kept your same old code. 
Additions are .eval() and torch.no_grad() and numbatches

**Comment by rasbt** [comment]
Hm that's weird, maybe you forgot to push the changes? It currently shows this in the diff pane:

screen shot 2019-01-22 at 10 05 40 am

**Comment by avinashsai** [comment]
Any further changes to be made?

**Comment by avinashsai** [comment]
Hi,
Can I know the status for this PR?

**Comment by rasbt** [comment]
This looks fine to me, however, could you change

- `return  logits,probas` back to `return logits, probas` to adhere with PEP8 style?
- instead of defining `numbatches = len(train_loader)` you can also use `len(train_loader)` which makes this probably easier to read (a reader wouldn't have to scroll up again to see where it's defined)

**Comment by avinashsai** [comment]
Hi,
I updated notebooks as per your suggestions. Please review the changes

**Comment by rasbt** [comment]
This looks great. Many thanks!
